sass/cd/cd-header/_cd-language-switcher.scss

```diff
@@ -6,123 +6,203 @@
+}
 
-  .cd-language-switcher & {
-    right: 0;
-    left: auto;
-    width: 125px;
+.js[dir="rtl"] .cd-language-switcher__dropdown {
+  right: auto;
+  left: 0;
+}
+
+.cd-language-switcher__dropdown li a {
+  position: relative;
+  display: block;
+  height: var(--cd-global-header-height);
+  text-decoration: none;
+  color: var(--cd-white);
+  line-height: var(--cd-global-header-height);
+  padding-inline: 0.75rem;
+
+  &:hover,
+  &:focus {
+    text-decoration: underline;
   }
 
-  li {
-    display: block;
-    float: none;
+  &:focus {
+    outline: 3px solid var(--brand-primary--light);
   }
 
-  li a {
-    @extend .cd-user-menu__dropdown-link;
+  &.is-active {
+    font-weight: 700;
+  }
+}
 
-    color: var(--cd-white);
+// JS disabled.
+// Language Switcher list.
+.no-js .cd-language-switcher__dropdown {
+  /* Unset the dropdown styles */
+  position: relative;
+  right: unset;
+  left: unset;
+  display: flex;
+  flex-wrap: wrap;
+  padding: 0;
+  background: transparent;
+  box-shadow: none;
+}
 
-    &:hover,
-    &:focus {
-      text-decoration: underline;
-    }
+// Pipe separator.
+.no-js .cd-language-switcher__dropdown li a::before {
+  position: absolute;
+  top: 50%;
+  left: -1px;
+  width: 1px;
+  height: calc(var(--cd-global-header-height) / 3);
+  content: "";
+  transform: translateY(-50%);
+  background: var(--brand-grey);
+
+  [dir="rtl"] & {
+    right: -1px;
+    left: auto;
+  }
+}
 
-    &:focus {
-      outline: 3px solid var(--brand-primary--light);
-    }
+// No pipe on first child.
+.no-js .cd-language-switcher__dropdown li:first-child a::before {
+  background: transparent;
+}
+
+// If there are menus in the Global Header
+// and the Language Switcher is displayed in full,
+// separate with pipes and margins.
+.cd-global-header__language-switcher + .cd-global-header__user-menu {
+  position: relative;
+  // Margin value matches the padding of the other buttons in the Global Header
+  // and takes into account the right padding from the last Language Item.
+  @include desktop {
+    margin-inline-start: 1rem;
   }
 }
 
-.cd-user-menu__dropdown-link {
-  display: block;
-  padding: 12px 0;
-  text-decoration: none;
+// Pipe separator between Language Switcher and following menu.
+.js {
+  .cd-global-header__language-switcher + .cd-global-header__user-menu::before {
+    position: absolute;
+    top: 50%;
+    left: -1px;
+    width: 2px;
+    height: calc(var(--cd-global-header-height) / 3);
+    content: "";
+    transform: translateY(-50%);
+    background: var(--brand-grey);
+  }
 
-  &.is-active {
+  &[dir="rtl"] .cd-global-header__language-switcher + .cd-global-header__user-menu::before {
+    right: -1px;
+    left: auto;
+  }
+}
+
+// Adjust breakpoints for when the Language Switcher displays as a dropdown.
+// This depends on other elements present in the .cd-global-header__actions.
+//
+// Copy the following rules into the sub theme and override breakpoint values.
```

Comment:
This corresponds with https://github.com/UN-OCHA/common-design-site/pull/225 where there is a component added to the sub theme to override the Language Switcher, to demonstrate how we can adjust the breakpoint at which the Language Switcher displays in full, per implementation, based on the other elements in the Global Header.

Reply:
Thanks for highlighting this. It took me a while to find these styles and reset in all the places I needed. By coincidence, FR/ES/AR languages produced overflow at the default 768px breakpoint when I tested locally using the menus enabled on our CD Demo site (lang x6, help, user).

Reply:
The block has been reduced quite a bit by eliminating pseudo-element pipes. I did leave the component in the CD Site PR but also copied it to the sub-theme starterkit (disabled by default) so that it's immediately available to override on any site.

Once we finally get rid of Sass and have each component defined as a Drupal library, we can change our commented-out examples in the theme info file to use Language Switcher so that overriding the breakpoint is a couple keystrokes.